packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/helpers.ts

```diff
@@ -218,11 +218,15 @@ export function buildRemoveAliasActions(
 /**
  * Given a document, creates a valid body to index the document using the Bulk API.
  */
-export const createBulkIndexOperationTuple = (doc: SavedObjectsRawDoc): BulkIndexOperationTuple => {
+export const createBulkIndexOperationTuple = (
+  doc: SavedObjectsRawDoc,
+  typeIndexMap: Record<string, string> = {}
+): BulkIndexOperationTuple => {
   return [
     {
       index: {
         _id: doc._id,
+        ...(typeIndexMap[doc._source.type] && { _index: typeIndexMap[doc._source.type] }),
```

Comment:
We no longer reindex to one single temporary index, but potentially to different ones, depending on the type.